pkg/resmap/resmap.go

```diff
@@ -92,10 +92,18 @@ type ResMap interface {
 	// who's CurId is matched by the argument.
 	GetMatchingResourcesByCurrentId(matches IdMatcher) []*resource.Resource
 
+	// GetMatchingResourcesByCurrentIdAndCtx returns the resources
+	// who's CurId is matched by the argument.
+	GetMatchingResourcesByCurrentIdAndCtx(matches IdMatcher, rctxm resource.ResCtxMatcher) []*resource.Resource
```

Comment:
Instead of adding two functions to `ResMap`, you can add a filter function for `[]*resource.Resource`. Then call this filter function after `GetMatchingResourcesByCurrentId`.

Reply:
@Liujingfang1 Done.  Thanks to your remark. Was able to figure out a better way to filter like you advised during the candidate selecction.